docs/driverapi.md

```diff
@@ -0,0 +1,398 @@
+For an Idemix identity (or pseudonym), it contains the secrets to de-anonymise the pseudonym.
+The `info` field in the `IdentitySigners` table remains empty as well as `token_metadata` and `token_metadata_audit_info`.
+The `KeyStore` table contains a row whose `key` field gets the value of the SKI of the cryptographic key stored and value is the cryptographic key itself.
+
+#### 3.2.1.4 Identity and Access Management
+
+The Token-SDK uses identities to establish trust and control access within the system.
+These identities are like digital passports that verify who a party is and what actions they're authorized to perform.
+Therefore, the Token-SDK's identity service offers:
+1. A way to define these identities,
+2. A way to generate and verify digital signatures valid under these identities, and
+3. A way to marshal/unmarshal identities and signatures.
+
+#### Roles as Teams
+
+Within the identity service, long-term identities are grouped into roles.
+Imagine these roles as teams with specific permissions.
+Here are some key roles:
+
+* **Issuers:** Like a mint that creates coins, issuers have the power to create new tokens.
+* **Owners:** Owners hold tokens, just like possessing a digital asset.
+* **Auditors:** These act as financial inspectors, overseeing token requests and ensuring proper use.
+* **Certifiers:** They verify the existence and legitimacy of specific tokens, similar to checking identification.
+  This role is used only by certain token drivers that support the so called `graph hiding`.
+
+Recall that long-term identities are stored in the `IdentityConfigurations`.
+The corresponding secrets are stored either in the `KeyStore` table or an external key store, if configured.
+
+####  Identity Options: Passports and Beyond
+
+Identities come in different forms.
+
+A common choice is an `X.509 certificate`, a secure electronic passport that links a real-world entity (website, organization, or person) to a public key using a digital signature.
+This certificate uniquely identifies the entity it represents.
+
+Another option is `anonymous credentials`, which allow users to prove they have certain attributes (like age or qualifications) without revealing their entire identity.
+Imagine showing an ID that only displays relevant information, not your full details.
+This is particularly useful for protecting user privacy.
+
+####  Roles and Wallets: Managing Access
+
+Importantly, roles can contain different identity types.
+These roles then act as the foundation for creating wallets.
+A wallet is like a digital vault that stores a long-term identity (the main key) and any credentials derived from it, if supported.
+Different wallet types provide different functionalities.
+For example, an issuer wallet lets you see a list of issued tokens, while an owner wallet shows unspent tokens.
+All these wallets have though a minimum common denominator:
+Given a wallet, you, as the developer, can derive identities to be used in different contexts.
```

Comment:
```suggestion
Given a wallet, identities can be derived for use in different contexts.
```

Regarding, till now the document used a formal tone, and now it has a different tutorial like tone ("you, as the developer,")